src/components/scenes/Fio/FioCreateHandleScene.tsx

```diff
@@ -37,128 +34,119 @@ const asRegisterFailedRes = asObject({
   error: asString
 })
 
+const asFreeFioDomain = asObject({
+  domain: asString,
+  free: asValue(true)
+})
+
 export const FioCreateHandleScene = ({ navigation }: Props) => {
   const theme = useTheme()
   const styles = getStyles(theme)
   const dispatch = useDispatch()
   const accountUserName = useSelector(state => state.core.account.username)
-  const domainStr = `@${config.appName.toLowerCase()}`
+  const fioPlugin = useSelector(state => state.core.account.currencyConfig.fio)
+
+  const { freeRegApiToken, freeRegRefCode } = typeof ENV.FIO_INIT === 'object' ? ENV.FIO_INIT : { freeRegApiToken: null, freeRegRefCode: null }
```

Comment:
Clean this up to

```
const { freeRegApiToken, freeRegRefCode } = typeof ENV.FIO_INIT === 'object' ? ENV.FIO_INIT : {}
```

Reply:
If these are null, immediately pop(). In fact we shouldn't even show the first modal if these are undefined